src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import { useParams } from 'react-router-dom';
+
+const getPeopleWithParents = (people: Person[]) => {
+  return people.map(person => {
+    return {
+      ...person,
+      father: people.find(f => f.name === person.fatherName),
+      mother: people.find(m => m.name === person.motherName),
+    };
+  });
+};
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const { personSlug } = useParams<string>();
+
+  useEffect(() => {
+    setLoading(true);
+    setError(false);
+
+    getPeople()
+      .then(p => setPeople(getPeopleWithParents(p)))
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.